### PR TITLE
Splitted application into different modules

### PR DESCRIPTION
**Pull request by achehrynets**
https://trello.com/c/hjes74LX/140-divide-project-on-modules

### CODE REVIEW

File: service/pom.xml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            <groupId>com.softserve</groupId>
+            <artifactId>persistence</artifactId>
+            <version>1.0-SNAPSHOT</version>
+            <scope>compile</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.softserve</groupId>
+            <artifactId>shared</artifactId>
+            <version>1.0-SNAPSHOT</version>
+            <scope>compile</scope>
```
Comment (achehrynets) [inline service/pom.xml]: It is the default scope of a maven dependency.
I would recommend removing it.
https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html

File: service/src/main/java/com/softserve/borda/services/impl/BoardListServiceImpl.java [hunk 2]
```diff
@@ -38,14 +36,14 @@ public BoardList deleteTicketFromBoardList(BoardList boardList, Ticket ticket) {
             ticketService.deleteTicketById(ticket.getId());
             return boardListRepository.save(boardList);
         } catch (Exception e) {
-            throw new CustomFailedToDeleteEntityException(e.getMessage());
+            throw new com.softserve.borda.exceptions.CustomFailedToDeleteEntityException(e.getMessage());
```
Comment (achehrynets) [inline service/src/main/java/com/softserve/borda/services/impl/BoardListServiceImpl.java]: Can you explain a reason of adding relative path to CustomFailedToDeleteEntityException class?

File: service/src/main/java/com/softserve/borda/services/impl/BoardListServiceImpl.java [hunk 3]
```diff
@@ -38,14 +36,14 @@ public BoardList deleteTicketFromBoardList(BoardList boardList, Ticket ticket) {
...
         }
     }
 
     @Override
     public BoardList getBoardListById(Long id) {
         return boardListRepository.findById(id)
-                .orElseThrow(() -> new CustomEntityNotFoundException(BoardList.class));
+                .orElseThrow(() -> new com.softserve.borda.exceptions.CustomEntityNotFoundException(BoardList.class));
```
Comment (achehrynets) [inline service/src/main/java/com/softserve/borda/services/impl/BoardListServiceImpl.java]: I would recommend using of ExceptionHandler for CustomEntityNotFoundException (https://www.baeldung.com/exception-handling-for-rest-with-spring)

File: application/src/main/resources/application.properties [hunk 4]
```diff
@@ -2,9 +2,9 @@ server.port=9090
 
 spring.jpa.database=POSTGRESQL
 spring.datasource.platform=postgres
-spring.datasource.url=${DATASOURCE_URL}
-spring.datasource.username=${DB_USER}
-spring.datasource.password=${DB_PASSWORD}
+spring.datasource.url=jdbc:postgresql://localhost:5432/borda
```
Comment (achehrynets) [inline application/src/main/resources/application.properties]: Do not expose credentials.

File: controller/src/main/java/com/softserve/borda/config/jwt/JwtProvider.java [hunk 5]
```diff
@@ -19,7 +15,7 @@
     private String jwtSecret;
 
     public String generateToken(String login) {
-        Date date = Date.from(LocalDate.now().plusDays(15).atStartOfDay(ZoneId.systemDefault()).toInstant());
+//        Date date = Date.from(LocalDate.now().plusDays(15).atStartOfDay(ZoneId.systemDefault()).toInstant());
```
Comment (achehrynets) [inline controller/src/main/java/com/softserve/borda/config/jwt/JwtProvider.java]: Remove comment

File: service/src/test/resources/application.properties [hunk 6]
```diff
@@ -0,0 +1,13 @@
+server.port=9090
+
+spring.jpa.database=POSTGRESQL
+spring.datasource.platform=postgres
+spring.datasource.url=jdbc:postgresql://localhost:5432/borda
+spring.datasource.username=borda
```
Comment (achehrynets) [inline service/src/test/resources/application.properties]: do not expose credentials

**Review by achehrynets — COMMENTED** [review]
Left some comments

File: service/src/main/java/com/softserve/borda/services/impl/BoardListServiceImpl.java (hunk 2 above)
Comment (ChernykhVyacheslav) [reply service/src/main/java/com/softserve/borda/services/impl/BoardListServiceImpl.java]: Idea must have imported it this way after dividing project into modules and I missed it. Will fix it

File: controller/src/main/java/com/softserve/borda/config/jwt/JwtProvider.java (hunk 5 above)
Comment (ChernykhVyacheslav) [reply controller/src/main/java/com/softserve/borda/config/jwt/JwtProvider.java]: It is just that 'date' variable was never used. Now I uncommented it together with line where it should be used.

**Review by HorbatenkoYehor — APPROVED** [review]
+1

**Review by jules-n — APPROVED** [review]
Good work

**Pull request merged**